packages/webdriverio/src/commands/browser/switchWindow.ts

```diff
@@ -45,10 +45,14 @@ export async function switchWindow (
         throw new Error('Unsupported parameter for switchWindow, required is "string" or a RegExp')
     }
 
-    const currentWindow = await this.getWindowHandle()
+    let currentWindow
+
+    try {
+        currentWindow = await this.getWindowHandle()
+    } catch {/* In case where the current window is closed but there are still other open windows */}
 
     // is the matcher a window handle, and are we in the right window already?
-    if (typeof matcher === 'string' && currentWindow === matcher) {
+    if (currentWindow && typeof matcher === 'string' && currentWindow === matcher) {
```

Comment:
This doesn't seem to be needed.
```suggestion
    if (typeof matcher === 'string' && currentWindow === matcher) {
```